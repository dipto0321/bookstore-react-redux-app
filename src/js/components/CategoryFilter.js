import React from 'react';
import PropTypes from 'prop-types';
import categories from '../data/bookCategories';

const CategoryFilter = (props) => {
  const {
    filter,
    handleChange,
  } = props;
  return (
    <select
      value={filter}
      onChange={e => handleChange(e.target.value)}
    >
      {
    ['All'].concat(categories).map(category => (
      <option
        key={category}
        value={category}
      >
        { category }
      </option>
    ))
  }
    </select>
  );
};

CategoryFilter.propTypes = {
  filter: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default CategoryFilter;
