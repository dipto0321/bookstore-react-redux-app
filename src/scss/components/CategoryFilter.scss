.category-container {
  display: flex;
  height: $medium;
  width: $large;
}