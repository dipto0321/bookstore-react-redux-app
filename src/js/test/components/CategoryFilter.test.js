import React from 'react';
import { shallow } from 'enzyme';
import CategoryFilter from '../../components/CategoryFilter';

describe('CategoryFilter', () => {
  let wrapper;
  let filter;
  let handleChange;

  beforeEach(() => {
    filter = 'All';
    handleChange = jest.fn();
    wrapper = shallow(<CategoryFilter filter={filter} handleChange={handleChange} />);
  });

  test('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('handleChange should be called with new category', () => {
    const actionCategory = 'Action';
    wrapper.find('select').simulate('change', {
      target: { value: actionCategory },
    });
    expect(handleChange).toHaveBeenLastCalledWith(actionCategory);
  });
});
