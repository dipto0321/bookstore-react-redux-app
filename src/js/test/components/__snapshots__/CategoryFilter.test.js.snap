// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter it should render correctly 1`] = `
<div
  className="category-container"
>
  <h2
    className="filter-title"
  >
    Filter category: 
  </h2>
  <select
    className="category-select form-select"
    onChange={[Function]}
    value="All"
  >
    <option
      key="All"
      value="All"
    >
      All
    </option>
    <option
      key="Action"
      value="Action"
    >
      Action
    </option>
    <option
      key="Biography"
      value="Biography"
    >
      Biography
    </option>
    <option
      key="History"
      value="History"
    >
      History
    </option>
    <option
      key="Horror"
      value="Horror"
    >
      Horror
    </option>
    <option
      key="Kids"
      value="Kids"
    >
      Kids
    </option>
    <option
      key="Learning"
      value="Learning"
    >
      Learning
    </option>
    <option
      key="Sci-Fi"
      value="Sci-Fi"
    >
      Sci-Fi
    </option>
  </select>
</div>
`;
